client/src/components/Verejne/Map/index.js

```diff
@@ -80,11 +85,27 @@ export default compose(
       zoomToLocation,
     }
   ),
+  withRouter,
   withDataProviders(({entitiesUrl}) => [entitiesProvider(entitiesUrl)]),
   withHandlers({
-    onChange: ({entitiesUrl, setMapOptions, refetch}) => (options) => {
-      setMapOptions(options)
-    },
+    onChange: (props) => (options) => {
+      console.log(props)
+      props.setMapOptions(options)
```

Comment:
you definitely don't want to do this debounced - you'll probably want to call a debounced function from inside onChange